host/src/server/api/v1/proof.rs

```diff
@@ -108,21 +141,22 @@ async fn handle_proof(
         taiko_chain_spec.clone(),
         proof_request.clone(),
     );
-    let input = if let Some(cached_input) = cached_input {
-        debug!("Using cached input");
-        cached_input
-    } else {
-        memory::reset_stats();
-        let measurement = Measurement::start("Generating input...", false);
-        let provider = RpcBlockDataProvider::new(
-            &taiko_chain_spec.rpc.clone(),
-            proof_request.block_number - 1,
-        )?;
-        let input = raiko.generate_input(provider).await?;
-        let input_time = measurement.stop_with("=> Input generated");
-        observe_prepare_input_time(proof_request.block_number, input_time, true);
-        memory::print_stats("Input generation peak memory used: ");
-        input
+    let provider = RpcBlockDataProvider::new(
+        &taiko_chain_spec.rpc.clone(),
+        proof_request.block_number - 1,
+    )?;
+    let input = match validate_cache_input(cached_input, &provider).await {
+        Ok(cache_input) => cache_input,
+        Err(_) => {
+            // no valid cache
+            memory::reset_stats();
+            let measurement = Measurement::start("Generating input...", false);
+            let input = raiko.generate_input(provider).await?;
+            let input_time = measurement.stop_with("=> Input generated");
+            observe_prepare_input_time(proof_request.block_number, input_time, true);
+            memory::print_stats("Input generation peak memory used: ");
+            input
+        }
```

Comment:
And here I'd probably just use an `if let` statement because we don't need the error so I'd do the following:
```rust
    let input = if let Ok(cache_input) = validate_cache_input(cached_input, &provider).await {
        cache_input
    } else {
        // no valid cache
        memory::reset_stats();
        let measurement = Measurement::start("Generating input...", false);
        let input = raiko.generate_input(provider).await?;
        let input_time = measurement.stop_with("=> Input generated");
        observe_prepare_input_time(proof_request.block_number, input_time, true);
        memory::print_stats("Input generation peak memory used: ");
        input
    };
```

Reply:
Yes, I used if let before, and changes after while, just in case of printing the error(seems useless now)...as so far I think they are the same, let me keep it :)